x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/supported_agents/java_runtime_attachment.tsx

```diff
@@ -0,0 +1,276 @@
+                },
+                java_attacher_discovery_rules: {
+                  type: 'yaml',
+                  value: encodeDiscoveryRulesYaml(
+                    runtimeAttachmentSettings.discoveryRules
+                  ),
+                },
+                java_attacher_agent_version: {
+                  type: 'text',
+                  value: runtimeAttachmentSettings.version,
+                },
+              },
+            },
+            ...newPolicy.inputs.slice(apmInputIdx + 1),
+          ],
+        },
+      });
+      setIsDirty(true);
+    },
+    [newPolicy, onChange]
+  );
+
+  const apmVars = useMemo(() => getApmVars(newPolicy), [newPolicy]);
+
+  return (
+    <RuntimeAttachment
+      operationTypes={[
+        {
+          operation: {
+            value: 'include',
+            label: i18n.translate(
+              'xpack.apm.fleetIntegration.javaRuntime.operationType.include',
+              { defaultMessage: 'Include' }
+            ),
+          },
+          types: includeOptions,
+        },
+        {
+          operation: {
+            value: 'exclude',
+            label: i18n.translate(
+              'xpack.apm.fleetIntegration.javaRuntime.operationType.exclude',
+              { defaultMessage: 'Exclude' }
+            ),
+          },
+          types: excludeOptions,
+        },
+      ]}
+      onChange={onChangePolicy}
+      toggleDescription={i18n.translate(
+        'xpack.apm.fleetIntegration.javaRuntime.toggleDescription',
+        {
+          defaultMessage:
+            'Attach the Java agent to running and starting Java applications.',
+        }
+      )}
+      discoveryRulesDescription={
+        <FormattedMessage
+          id="xpack.apm.fleetIntegration.javaRuntime.discoveryRulesDescription"
+          defaultMessage="For every running JVM, the discovery rules are evaluated in the order they are provided. The first matching rule determines the outcome. Learn more in the {docLink}."
+          values={{
+            docLink: (
+              <a
+                href="https://www.elastic.co/guide/en/apm/agent/java/current/setup-attach-cli.html"
+                target="_blank"
+              >
+                {i18n.translate(
+                  'xpack.apm.fleetIntegration.javaRuntime.discoveryRulesDescription.docLink',
+                  { defaultMessage: 'docs' }
+                )}
+              </a>
+            ),
+          }}
+        />
+      }
+      showUnsavedWarning={isDirty}
+      initialIsEnabled={apmVars?.java_attacher_enabled?.value}
+      initialDiscoveryRules={decodeDiscoveryRulesYaml(
+        apmVars?.java_attacher_discovery_rules?.value ?? '[]\n',
+        [initialDiscoveryRule]
+      )}
+      selectedVersion={
+        apmVars?.java_attacher_agent_version?.value || versions[0]
+      }
+      versions={versions}
+    />
+  );
+}
+
+const initialDiscoveryRule = {
+  operation: 'include',
+  type: 'vmarg',
+  probe: 'elastic.apm.attach=true',
```

Comment:
@felixbarny is this OK, or do you prefer `elastic.apm.agent.attach`?

Reply:
I don't have a strong opinion. LGTM.

Reply:
OK, let's stick to that then.
I only asked because we use `elastic.apm.agent.attach` in our [docs example](https://www.elastic.co/guide/en/apm/agent/java/current/setup-attach-cli.html#setup-attach-cli-usage).